pkg/meta/sql.go

```diff
@@ -2231,6 +2259,73 @@ func (m *dbMeta) doGetParents(ctx Context, inode Ino) map[Ino]int {
 	return ps
 }
 
+func (m *dbMeta) doIncreDirUsage(ctx Context, ino Ino, space int64, inodes int64) error {
+	if space == 0 && inodes == 0 {
+		return nil
+	}
+	return m.txn(func(s *xorm.Session) error {
+		table := m.db.GetTableMapper().Obj2Table("dirUsage")
+		usedSpaceColumn := m.db.GetColumnMapper().Obj2Table("UsedSpace")
+		usedInodeColumn := m.db.GetColumnMapper().Obj2Table("UsedInodes")
+		sql := fmt.Sprintf(
+			"update `%s` set `%s` = `%s` + ?, `%s` = `%s` + ? where `inode` = ?",
+			table,
+			usedSpaceColumn, usedSpaceColumn,
+			usedInodeColumn, usedInodeColumn,
+		)
+		ret, err := s.Exec(sql, space, inodes, ino)
+		if err != nil {
+			return err
+		}
+		if n, _ := ret.RowsAffected(); n > 0 {
+			return nil
+		}
+
+		totalSpace, totalInodes, countErr := m.countDirUsage(ctx, ino)
```

Comment:
maybe put it outside of the txn in case the directory is huge.

Reply:
Done